daemon/algod/api/server/v2/handlers.go

```diff
@@ -741,17 +741,8 @@ func (v2 *Handlers) WaitForBlock(ctx echo.Context, round uint64) error {
 	return v2.GetStatus(ctx)
 }
 
-func (v2 *Handlers) checkNodeAndDecodeTxGroup(ctx echo.Context) ([]transactions.SignedTxn, error) {
-	stat, err := v2.Node.Status()
-	if err != nil {
-		return nil, internalError(ctx, err, errFailedRetrievingNodeStatus, v2.Log)
-	}
-	if stat.Catchpoint != "" {
-		// node is currently catching up to the requested catchpoint.
-		return nil, serviceUnavailable(ctx, fmt.Errorf("RawTransaction failed as the node was catchpoint catchuping"), errOperationNotAvailableDuringCatchup, v2.Log)
-	}
-	proto := config.Consensus[stat.LastVersion]
-
+// decodeTxGroup attempts to decode a request body containing a transaction group.
+func decodeTxGroup(ctx echo.Context, maxTxGroupSize int) ([]transactions.SignedTxn, error) {
```

Comment:
nit: instead of accepting an `echo.Context`, accepting an `io.Reader` (the caller would have to pass `ctx.Request().Body`) would be more general, and it would prevent this function from accidentally doing more than it should with the request